src/kernel/src/net/socket.rs

```diff
@@ -23,7 +26,7 @@ impl Socket {
     pub fn new(
         domain: i32,
         ty: i32,
-        proto: i32,
+        proto: Option<NonZeroI32>,
```

Comment:
Why this one is non-zero? What I saw in the `socreate` is it have a path for zero proto.

Reply:
It's  not NonZero, it's Option<NonZero>. The reason I did this is because It looks like 0 is a special-case value that's treated differently. Sound kind of like an option to me.
<img width="367" alt="image" src="https://github.com/obhq/obliteration/assets/48620541/3e07ff07-dbe2-4aa1-8cd1-6db94736a802">
<img width="481" alt="image" src="https://github.com/obhq/obliteration/assets/48620541/3c43f0be-4bc6-445c-a3b7-15a7c4346742">

Reply:
Then this looking good.